### ISSUE TITLE
evluation different?

### ISSUE THREAD

**Issue by ZHUXUHAN** [thunlp/VisualDS#7]
In evaluation, weakly supervised trained model uses logits rather than softmax normalized scores for relation triplets ranking.
hi, why you use logits rather than softmax normalized scores?
i just meet the problems.

**Comment by waxnkw** [comment]
Because in the weakly supervised setting, the model is trained with BCE Loss rather than Cross Entropy. In this condition, given an object pair, the model may predict multiple relations with a high score. If we apply softmax, the score of each relation will all be low. 

For example, given (human, ?, bike), the possible labels in the KB contain `on` and `ride`. In this condition, the model predicts the logits as (on, 9.23) and (ride, 10.21). If softmax is applied, both scores of `on` and `ride` will be around 0.5. Given another object pair (human, ?, water), and the possible labels in the KB only contain `drink`. After softmax, the relation `drink` will be over 0.9, which will lead to the (human, ride, bike) being ranked much lower than (human, drink, water). However, both predictions are correct. Thus, we use logits rather than softmax.
